File: Medium/InsertInterval.java
class Solution {
    public int[][] insert(int[][] intervals, int[] newInterval) {
        List<List<Integer>> list = new ArrayList<>();
        
        int i = 0; 
        int n = intervals.length;
        int index = 0;
        
        
        while(i < n){
            if(newInterval[0] > intervals[i][1]){
                list.add(intervals[i]);
                index++;
            }else if(intervals[i][0] > newInterval[1] ){
                list.add(intervals[i]);
            }else{
                newInterval[0] = Math.min(newInterval[0], intervals[i][0]);
                newInterval[1] = Math.max(newInterval[1], intervals[i][1]);
            }
            i++;
        }
        list.add(index, newInterval);
        return list;
        
    }
}

File: Medium/InsertInterval.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        if not intervals:
            return [newInterval]
        result = []
        n = len(intervals)
        i = 0
        index = 0
        
        while i < n:
            if newInterval[0] > intervals[i][1]:
                result.append(intervals[i])
                index += 1
            elif newInterval[1] < intervals[i][0]:
                result.append(intervals[i])
            else:
                newInterval[0] = min(newInterval[0], intervals[i][0])
                newInterval[1] = max(newInterval[1], intervals[i][1])
            i += 1 
        result.insert(index, newInterval)
        return result
                    